sentry/src/routes/campaign.rs

```diff
@@ -697,110 +697,144 @@ pub mod insert_events {
             (Some(leader), Some(follower)) => (leader, follower),
         };
 
-        let mut events_success = vec![];
-        for event in events.into_iter() {
-            let event_payout = {
-                // calculate earners payouts
-                let payout = get_payout(&app.logger, &campaign_context.context, &event, session)?;
-
-                match payout {
-                    Some((earner, payout)) => {
-                        let spending_result = spend_for_event(
-                            &app.pool,
-                            &app.campaign_remaining,
-                            &campaign_context.context,
-                            earner,
-                            leader,
-                            follower,
-                            payout,
-                        )
-                        .await;
+        let events_success = spend_for_events(
+            app,
+            &campaign_context.context,
+            events,
+            session,
+            leader,
+            follower,
+        )
+        .await?;
+
+        // Record successfully paid out events to Analytics
+        analytics_record_spawn(
+            app.pool.clone(),
+            app.logger.clone(),
+            campaign_context.clone(),
+            session.clone(),
+            events_success,
+        );
```

Comment:
Although we spawn a separate task, it does not improve the req/sec, probably because postgres is hitting a bottleneck when updating the same records.